unpublished/project_kit/lib/src/utils/token_storage/security_token_storage.dart

```diff
@@ -1,90 +1,82 @@
-}
-
-@freezed
-class TokenStorageState with _$TokenStorageState {
-  factory TokenStorageState({
-    required String refreshToken,
-    required String accessToken,
-  }) = _TokenStorageState;
-
-  factory TokenStorageState.fromJson(Map<String, dynamic> json) =>
-      _$TokenStorageStateFromJson(json);
-}
-
-/// Отвечает за управление и хранение токенов авторизации пользователя
-@Riverpod(keepAlive: true)
-class TokenStorage extends _$TokenStorage implements IRef {
-  static const _storage = FlutterSecureStorage();
-  static const _refreshKey = 'refreshToken';
-  static const _accessKey = 'accessToken';
-
-  @override
-  Future<TokenStorageState?> build() async {
-    try {
-      final refreshToken = await _storage.read(key: _refreshKey);
-      final accessToken = await _storage.read(key: _accessKey);
-
-      if (refreshToken == null || accessToken == null) return null;
-
-      return TokenStorageState(
-        refreshToken: refreshToken,
-        accessToken: accessToken,
-      );
-    } catch (e, s) {
-      logger.error(title: 'TokenStorage', error: e, stack: s);
-      return null;
-    }
-  }
-
-  void updateToken({
-    required String refreshToken,
-    required String accessToken,
-  }) {
-    _storage.write(key: _refreshKey, value: refreshToken);
-    _storage.write(key: _accessKey, value: accessToken);
-
-    setData(
-      TokenStorageState(
-        refreshToken: refreshToken,
-        accessToken: accessToken,
-      ),
-    );
-  }
-
-  void clearToken() {
-    _storage.delete(key: _refreshKey);
-    _storage.delete(key: _accessKey);
-
-    setData(null);
-  }
-}
+// ignore_for_file: avoid_positional_boolean_parameters
+
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:fresh_dio/fresh_dio.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import '../../../project_kit.dart';
+import '../../internal/logger/dev_kit_logger.dart';
+import '../notifier_async_utils/notifier_async_utils.dart';
+
+part 'security_token_storage.g.dart';
+
+@Riverpod(keepAlive: true)
+class UserChanged extends _$UserChanged {
+  @override
+  void build() {
+    ref.listen(securityTokenStorageProvider, (_, token) {
+      if (token.requireValue != null) ref.invalidateSelf();
+    });
+  }
+
+  @override
+  bool updateShouldNotify(void previous, void next) => true;
+}
+
+@Riverpod(keepAlive: true)
+bool userAuthorized(UserAuthorizedRef ref) {
+  final token = ref.watch(securityTokenStorageProvider);
+  return token.requireValue != null;
+}
+
+/// Отвечает за управление и хранение токенов авторизации пользователя
+@Riverpod(keepAlive: true)
+class SecurityTokenStorage extends _$SecurityTokenStorage
+    implements IRef, TokenStorage<AuthToken> {
+  static const _storage = FlutterSecureStorage();
+  static const _refreshKey = 'refreshToken';
+  static const _accessKey = 'accessToken';
+
+  @override
+  Future<AuthToken?> build() async {
+    await read();
+  }
```

Comment:
Мы же тут должны возращать AuthToken?

```suggestion
  Future<AuthToken?> build() async {
    final token =  await read();
    return token;
  }
```

Reply:
Конечно, е-мое, не заметил